lib/active_model_serializers/model.rb

```diff
@@ -0,0 +1,39 @@
+# ActiveModelSerializers::Model is a convenient
+# serializable class to inherit from when making
+# serializable non-activerecord objects.
+module ActiveModelSerializers
+  class Model
+    include ActiveModel::Model
+    include ActiveModel::Serializers::JSON
+
+    attr_reader :attributes
+
+    def initialize(attributes = {})
+      @attributes = attributes
+      super
+    end
+
+    # Defaults to the downcased model name.
+    def id
+      attributes.fetch(:id) { self.class.name.downcase }
+    end
+
+    # Defaults to the downcased model name and updated_at
+    def cache_key
+      attributes.fetch(:cache_key) { "#{self.class.name.downcase}/#{id}-#{updated_at.strftime("%Y%m%d%H%M%S%9N")}" }
+    end
+
+    # Defaults to the time the serializer file was modified.
+    def updated_at
+      attributes.fetch(:updated_at) { File.mtime(__FILE__) }
+    end
+
+    def read_attribute_for_serialization(key)
+      if key == :id || key == 'id'
+        attributes.fetch(key) { id }
```

Comment:
Well, it's used by the test suite like this... I'll double check tonight

B mobile phone

> On Oct 15, 2015, at 3:46 PM, L. Preston Sego III <notifications@github.com> wrote:
> 
> In lib/active_model_serializers/model.rb:
> 
> > +      attributes.fetch(:id) { self.class.name.downcase }
> > +    end
> > +
> > +    # Defaults to the downcased model name and updated_at
> > +    def cache_key
> > +      attributes.fetch(:cache_key) { "#{self.class.name.downcase}/#{id}-#{updated_at.strftime("%Y%m%d%H%M%S%9N")}" }
> > +    end
> > +
> > +    # Defaults to the time the serializer file was modified.
> > +    def updated_at
> > +      attributes.fetch(:updated_at) { File.mtime(__FILE__) }
> > +    end
> > +
> > +    def read_attribute_for_serialization(key)
> > +      if key == :id || key == 'id'
> > +        attributes.fetch(key) { id }
> Since this is just an example, is there anything wrong with just having attributes[key] ?
> 
> —
> Reply to this email directly or view it on GitHub.
>